src/Uns.ts

```diff
@@ -33,6 +33,37 @@ import UnsInternal from './UnsInternal';
 import Networking from './utils/Networking';
 import SupportedKeys from './config/resolver-keys.json';
 
+const ensureValidSourceConfig = (source: UnsSource): void => {
+  if (
+    !source.locations ||
+    !source.locations.Layer1 ||
+    !source.locations.Layer2
+  ) {
+    throw new ConfigurationError(ConfigurationErrorCode.NetworkConfigMissing, {
+      method: NamingServiceName.UNS,
+      config: !source.locations.Layer1 ? 'Layer1' : 'Layer2',
```

Comment:
If `source.locations` is null or undefined, this line will throw an error.
```suggestion
      config: !source?.locations?.Layer1 ? 'Layer1' : 'Layer2',
```